addons/stock/models/stock_picking.py

```diff
@@ -1496,6 +1496,19 @@ def action_open_label_layout(self):
                 'default_picking_quantity': 'picking'},
         }
 
+    def action_open_label_type(self):
+        if self.user_has_groups('stock.group_production_lot') and self.move_line_ids.lot_id:
```

Comment:
Should also work in receipt before validate with lot_name IMO

Reply:
We can also not support it but it's strange to have different behaviors before/after action_done

Reply:
We can maybe add it as a follow on task. Right now this uses only lot_id so we can re-use the existing lot/sn report actions (i.e. we pass the lot_id.ids). If we want to support lot_name also then we will have to create some sort of custom report that can handle both lot_ids + the lot_names while also still making them all print in the same 4x12 page together....